Sources/SwiftLexicalLookup/ScopeSyntax.swift

```diff
@@ -0,0 +1,109 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension SyntaxProtocol {
+  /// Returns all names that `for` refers to at this syntax node.
+  /// Optional configuration can be passed as `with` to customize the lookup behavior.
+  ///
+  /// - Returns: An array of `LookupResult` for name `for` at this syntax node,
+  /// ordered by visibility. If set to `nil`, returns all available names ordered by visibility.
```

Comment:
It took me a while to realize what `If set to nil` referred to. I think it would be good to re-staste `name` here.

```suggestion
  /// ordered by visibility. If `name` is set to `nil`, returns all available names ordered by visibility.
```